node v10 support?

When installing under node 10.3.0 I get these errors from node_modules\nan\nan.h(834)
```
error C2039: 'ForceSet': is not a member of 'v8::Object' (compiling source file ..\src\statement.cc)
```
they're the same as this issue: https://github.com/Level/leveldown/issues/448

I think the solution is to update nan to the latest version.  Could you do so?

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -33,7 +33,7 @@ aliases:
       # Install git-lfs (not part of the standard Docker images)
       - run: curl -sSL https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash && sudo apt-get update && sudo apt-get install -y git-lfs && git-lfs install --force && git-lfs pull
       # These are required for electron(-mocha)
-      - run: sudo apt-get install -y libgtk2.0-0 libxtst6 libxss1 libgconf2-4 libnss3 libasound2
+      - run: sudo apt-get install -y libgtk2.0-0 libgtk-3-dev libxtst6 libxss1 libgconf2-4 libnss3 libasound2
       - run: npm install --build-from-source
       - run: npm install --dev electron@$ELECTRON_VERSION
       - run: ./node_modules/.bin/electron-rebuild
@@ -152,7 +152,7 @@ workflows:
       - build-7.9.0
       - build-electron-2.0.4
       - build-electron-1.8.6
-      - build-electron-1.7.9
+      - build-electron-1.7.11
       - build-macos-10.3.0
       - build-macos-8.9.0
       - build-macos-electron-2.0.4

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -54,16 +54,16 @@ aliases:
       - run: git lfs pull
       - restore_cache:
           keys:
-          - v1-dependencies-osx-{{ checksum "package-lock.json" }}
+          - v1-dependencies-macos-{{ .Environment.CIRCLE_JOB }}-{{ checksum "package-lock.json" }}
           # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-osx-
+          - v1-dependencies-macos-{{ .Environment.CIRCLE_JOB }}
 
       - run: npm install --build-from-source
 
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-osx-{{ checksum "package-lock.json" }}
+          key: v1-dependencies-macos-{{ .Environment.CIRCLE_JOB }}-{{ checksum "package-lock.json" }}
 
       - run: npm run test
       - run: ./node_modules/.bin/electron-rebuild

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,6 +3,7 @@ aliases:
     name: Publish
     command: |
       COMMIT_MESSAGE=`git log --format=%s -n 1 $CIRCLE_SHA1`
+      ./node_modules/.bin/node-pre-gyp rebuild package testpackage $PRE_GYP_ARGS
       if [[ ${COMMIT_MESSAGE} =~ "[publish binary]" ]]; then
         ./node_modules/.bin/node-pre-gyp package testpackage $PRE_GYP_ARGS
         ./node_modules/.bin/node-pre-gyp publish $PRE_GYP_ARGS
@@ -35,7 +36,6 @@ aliases:
       # These are required for electron(-mocha)
       - run: sudo apt-get install -y libgtk2.0-0 libgtk-3-dev libxtst6 libxss1 libgconf2-4 libnss3 libasound2
       - run: npm install --build-from-source
-      - run: npm install --dev electron@$ELECTRON_VERSION
       - run: ./node_modules/.bin/electron-rebuild
       - run: npm run pretest
       - run: xvfb-run ./node_modules/.bin/electron-mocha --timeout 480000
@@ -59,7 +59,6 @@ aliases:
           - v1-dependencies-osx-
 
       - run: npm install --build-from-source
-      - run: npm install --dev electron@$ELECTRON_VERSION
 
       - save_cache:
           paths:
@@ -97,20 +96,17 @@ jobs:
   build-electron-2.0.4:
     <<: *common-electron-linux
     environment:
-      ELECTRON_VERSION: 2.0.4
-      PRE_GYP_ARGS: --runtime=electron --target=2.0.4
+      PRE_GYP_ARGS: --runtime=electron --target=2.0.4 --disturl=https://atom.io/download/electron
 
   build-electron-1.8.6:
     <<: *common-electron-linux
     environment:
-      ELECTRON_VERSION: 1.8.6
-      PRE_GYP_ARGS: --runtime=electron --target=1.8.6
+      PRE_GYP_ARGS: --runtime=electron --target=1.8.6 --disturl=https://atom.io/download/electron
 
   build-electron-1.7.11:
     <<: *common-electron-linux
     environment:
-      ELECTRON_VERSION: 1.7.11
-      PRE_GYP_ARGS: --runtime=electron --target=1.7.9
+      PRE_GYP_ARGS: --runtime=electron --target=1.7.9 --disturl=https://atom.io/download/electron
 
 
   build-macos-10.3.0:
@@ -127,20 +123,17 @@ jobs:
   build-macos-electron-2.0.4:
     <<: *common-macos
     environment:
-      ELECTRON_VERSION: 2.0.4
-      PRE_GYP_ARGS: --runtime=electron --target=2.0.4
+      PRE_GYP_ARGS: --runtime=electron --target=2.0.4 --disturl=https://atom.io/download/electron
 
   build-macos-electron-1.8.6:
     <<: *common-macos
     environment:
-      ELECTRON_VERSION: 1.8.6
-      PRE_GYP_ARGS: --runtime=electron --target=1.8.6
+      PRE_GYP_ARGS: --runtime=electron --target=1.8.6 --disturl=https://atom.io/download/electron
 
   build-macos-electron-1.7.11:
     <<: *common-macos
     environment:
-      ELECTRON_VERSION: 1.7.11
-      PRE_GYP_ARGS: --runtime=electron --target=1.7.11
+      PRE_GYP_ARGS: --runtime=electron --target=1.7.11 --disturl=https://atom.io/download/electron
 
 workflows:
   version: 2

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -35,11 +35,42 @@ aliases:
       # These are required for electron(-mocha)
       - run: sudo apt-get install -y libgtk2.0-0 libxtst6 libxss1 libgconf2-4 libnss3 libasound2
       - run: npm install --build-from-source
-      - run: npm install electron@$ELECTRON_VERSION
+      - run: npm install --dev electron@$ELECTRON_VERSION
       - run: ./node_modules/.bin/electron-rebuild
       - run: npm run pretest
       - run: xvfb-run ./node_modules/.bin/electron-mocha --timeout 480000
       - run: *publish-command
+
+
+  - &common-macos
+    macos:
+      xcode: "9.4.0" # Node 10.3.0
+    environment:
+      ELECTRON_VERSION: 2.0.4
+    steps:
+      - checkout
+      - run: brew install git-lfs openssl
+      - run: git lfs install
+      - run: git lfs pull
+      - restore_cache:
+          keys:
+          - v1-dependencies-osx-{{ checksum "package-lock.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-osx-
+
+      - run: npm install --build-from-source
+      - run: npm install --dev electron@$ELECTRON_VERSION
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-osx-{{ checksum "package-lock.json" }}
+
+      - run: npm run test
+      - run: ./node_modules/.bin/electron-rebuild
+      - run: ./node_modules/.bin/electron-mocha --timeout 480000
+      - run:
+          <<: *publish-command
 
 version: 2
 jobs:
@@ -75,40 +106,41 @@ jobs:
       ELECTRON_VERSION: 1.8.6
       PRE_GYP_ARGS: --runtime=electron --target=1.8.6
 
-  build-electron-1.7.9:
+  build-electron-1.7.11:
     <<: *common-electron-linux
     environment:
-      ELECTRON_VERSION: 1.7.9
+      ELECTRON_VERSION: 1.7.11
       PRE_GYP_ARGS: --runtime=electron --target=1.7.9
 
 
-  build-macos:
+  build-macos-10.3.0:
+    <<: *common-macos
+    macos:
+      xcode: "9.4.0" # Node 10.3.0
+
+  build-macos-8.9.0:
+    <<: *common-macos
     macos:
       xcode: "9.1.0" # Node 8.9.0
 
-    steps:
-      - checkout
-      - run: brew install git-lfs openssl
-      - run: git lfs install
-      - run: git lfs pull
-      - restore_cache:
-          keys:
-          - v1-dependencies-osx-{{ checksum "package-lock.json" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-osx-
 
-      - run: npm install --build-from-source
+  build-macos-electron-2.0.4:
+    <<: *common-macos
+    environment:
+      ELECTRON_VERSION: 2.0.4
+      PRE_GYP_ARGS: --runtime=electron --target=2.0.4
 
-      - save_cache:
-          paths:
-            - node_modules
-          key: v1-dependencies-osx-{{ checksum "package-lock.json" }}
+  build-macos-electron-1.8.6:
+    <<: *common-macos
+    environment:
+      ELECTRON_VERSION: 1.8.6
+      PRE_GYP_ARGS: --runtime=electron --target=1.8.6
 
-      - run: npm run test
-      - run: ./node_modules/.bin/electron-rebuild
-      - run: ./node_modules/.bin/electron-mocha --timeout 480000
-      - run:
-          <<: *publish-command
+  build-macos-electron-1.7.11:
+    <<: *common-macos
+    environment:
+      ELECTRON_VERSION: 1.7.11
+      PRE_GYP_ARGS: --runtime=electron --target=1.7.11
 
 workflows:
   version: 2
@@ -121,4 +153,8 @@ workflows:
       - build-electron-2.0.4
       - build-electron-1.8.6
       - build-electron-1.7.9
-      - build-macos
+      - build-macos-10.3.0
+      - build-macos-8.9.0
+      - build-macos-electron-2.0.4
+      - build-macos-electron-1.8.6
+      - build-macos-electron-1.7.11

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -44,9 +44,7 @@ aliases:
 
   - &common-macos
     macos:
-      xcode: "9.4.0" # Node 10.3.0
-    environment:
-      ELECTRON_VERSION: 2.0.4
+      xcode: "9.3.1" # Node 10.1.0
     steps:
       - checkout
       - run: brew install git-lfs openssl
@@ -109,10 +107,10 @@ jobs:
       PRE_GYP_ARGS: --runtime=electron --target=1.7.9 --disturl=https://atom.io/download/electron
 
 
-  build-macos-10.3.0:
+  build-macos-10.1.0:
     <<: *common-macos
     macos:
-      xcode: "9.4.0" # Node 10.3.0
+      xcode: "9.3.1" # Node 10.3.0
 
   build-macos-8.9.0:
     <<: *common-macos
@@ -146,7 +144,7 @@ workflows:
       - build-electron-2.0.4
       - build-electron-1.8.6
       - build-electron-1.7.11
-      - build-macos-10.3.0
+      - build-macos-10.1.0
       - build-macos-8.9.0
       - build-macos-electron-2.0.4
       - build-macos-electron-1.8.6

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,12 +1,17 @@
-version: 2
-jobs:
-  build-node-common: &common-build
-    docker:
-      # The docker section here is not used, but required
-      - image: circleci/node:8.9.3
+aliases:
+  - &publish-command
+    name: Publish
+    command: |
+      COMMIT_MESSAGE=`git log --format=%s -n 1 $CIRCLE_SHA1`
+      if [[ ${COMMIT_MESSAGE} =~ "[publish binary]" ]]; then
+        ./node_modules/.bin/node-pre-gyp package testpackage $PRE_GYP_ARGS
+        ./node_modules/.bin/node-pre-gyp publish $PRE_GYP_ARGS
+      else
+        echo "Not a tagged build - not publishing"
+      fi
 
+  - &common-build
     working_directory: ~/repo
-
     steps:
       - checkout
       # Install git-lfs (not part of the standard Docker images)
@@ -14,17 +19,34 @@ jobs:
       - run: npm install --build-from-source
 
       - run: npm run test
-      - run: &publish-command
-          name: Publish
-          command: |
-            COMMIT_MESSAGE=`git log --format=%s -n 1 $CIRCLE_SHA1`
-            if [[ ${COMMIT_MESSAGE} =~ "[publish binary]" ]]; then
-              ./node_modules/.bin/node-pre-gyp package testpackage $PRE_GYP_ARGS
-              ./node_modules/.bin/node-pre-gyp publish $PRE_GYP_ARGS
-            else
-              echo "Not a tagged build - not publishing"
-            fi
+      - run: *publish-command
 
+
+  - &common-electron-linux
+    docker:
+      - image: circleci/node:10.5.0
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      # Install git-lfs (not part of the standard Docker images)
+      - run: curl -sSL https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash && sudo apt-get update && sudo apt-get install -y git-lfs && git-lfs install --force && git-lfs pull
+      # These are required for electron(-mocha)
+      - run: sudo apt-get install -y libgtk2.0-0 libxtst6 libxss1 libgconf2-4 libnss3 libasound2
+      - run: npm install --build-from-source
+      - run: npm install electron@$ELECTRON_VERSION
+      - run: ./node_modules/.bin/electron-rebuild
+      - run: npm run pretest
+      - run: xvfb-run ./node_modules/.bin/electron-mocha --timeout 480000
+      - run: *publish-command
+
+version: 2
+jobs:
+  build-10.5.0:
+    <<: *common-build
+    docker:
+      - image: circleci/node:10.5.0
 
   build-9.3.0:
     <<: *common-build
@@ -41,26 +63,24 @@ jobs:
     docker:
       - image: circleci/node:7.9.0
 
-  build-electron:
-    docker:
-      - image: circleci/node:8.9.3
-
-    working_directory: ~/repo
+  build-electron-2.0.4:
+    <<: *common-electron-linux
     environment:
+      ELECTRON_VERSION: 2.0.4
+      PRE_GYP_ARGS: --runtime=electron --target=2.0.4
+
+  build-electron-1.8.6:
+    <<: *common-electron-linux
+    environment:
+      ELECTRON_VERSION: 1.8.6
+      PRE_GYP_ARGS: --runtime=electron --target=1.8.6
+
+  build-electron-1.7.9:
+    <<: *common-electron-linux
+    environment:
+      ELECTRON_VERSION: 1.7.9
       PRE_GYP_ARGS: --runtime=electron --target=1.7.9
 
-    steps:
-      - checkout
-      # Install git-lfs (not part of the standard Docker images)
-      - run: curl -sSL https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash && sudo apt-get update && sudo apt-get install -y git-lfs && git-lfs install --force && git-lfs pull
-      # These are required for electron(-mocha)
-      - run: sudo apt-get install -y libgtk2.0-0 libxtst6 libxss1 libgconf2-4 libnss3 libasound2
-      - run: npm install --build-from-source
-      - run: ./node_modules/.bin/electron-rebuild
-      - run: npm run pretest
-      - run: xvfb-run ./node_modules/.bin/electron-mocha --timeout 480000
-      - run:
-          <<: *publish-command
 
   build-macos:
     macos:
@@ -94,8 +114,11 @@ workflows:
   version: 2
   build_all:
     jobs:
+      - build-10.5.0
       - build-9.3.0
       - build-8.9.3
       - build-7.9.0
-      - build-electron
+      - build-electron-2.0.4
+      - build-electron-1.8.6
+      - build-electron-1.7.9
       - build-macos

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -44,7 +44,7 @@ aliases:
 
   - &common-macos
     macos:
-      xcode: "9.3.1" # Node 10.1.0
+      xcode: "9.1.0" # Node 8.9.0
     steps:
       - checkout
       - run: |
@@ -146,7 +146,8 @@ workflows:
       - build-electron-2.0.4
       - build-electron-1.8.6
       - build-electron-1.7.11
-      - build-macos-10.1.0
+      # Not working currently
+      # - build-macos-10.1.0
       - build-macos-8.9.0
       - build-macos-electron-2.0.4
       - build-macos-electron-1.8.6

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -47,7 +47,9 @@ aliases:
       xcode: "9.3.1" # Node 10.1.0
     steps:
       - checkout
-      - run: brew install git-lfs openssl
+      - run: |
+          export HOMEBREW_NO_AUTO_UPDATE=1
+          brew install git-lfs openssl || true
       - run: git lfs install
       - run: git lfs pull
       - restore_cache:

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -6,6 +6,12 @@ environment:
     - nodejs_version: 8
       platform: x86
       msvs_toolset: 14
+    - nodejs_version: 10
+      platform: x64
+      msvs_toolset: 14
+    - nodejs_version: 10
+      platform: x86
+      msvs_toolset: 14
 
 os: Visual Studio 2015
 

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "url": "git://github.com/journeyapps/node-sqlcipher.git"
   },
   "dependencies": {
-    "nan": "~2.7.0",
+    "nan": "^2.10.0",
     "node-pre-gyp": "^0.10.0"
   },
   "bundledDependencies": [

--- a/package.json
+++ b/package.json
@@ -45,9 +45,9 @@
   ],
   "devDependencies": {
     "aws-sdk": "^2.251.1",
-    "electron": "^1.8.2",
+    "electron": "^2.0.4",
     "electron-mocha": "^5.0.0",
-    "electron-rebuild": "^1.6.0",
+    "electron-rebuild": "^1.8.1",
     "eslint": "3.5.0",
     "mocha": "^5.2.0"
   },

--- a/scripts/build-appveyor.bat
+++ b/scripts/build-appveyor.bat
@@ -140,11 +140,21 @@ IF NOT "%CM%" == "%CM:[publish binary]=%" (ECHO publishing && CALL node_modules\
 IF %ERRORLEVEL% NEQ 0 GOTO ERROR
 
 :: Do the same for electron
-CALL node_modules\.bin\node-pre-gyp package %TOOLSET_ARGS% --runtime=electron --target=1.7.9
+CALL node_modules\.bin\node-pre-gyp rebuild package %TOOLSET_ARGS% --runtime=electron --target=1.7.11 --disturl=https://atom.io/download/electron
 SET CM=%APPVEYOR_REPO_COMMIT_MESSAGE%
-IF NOT "%CM%" == "%CM:[publish binary]=%" (ECHO publishing && CALL node_modules\.bin\node-pre-gyp --msvs_version=%msvs_version% publish %TOOLSET_ARGS% --runtime=electron --target=1.7.9) ELSE (ECHO not publishing)
+IF NOT "%CM%" == "%CM:[publish binary]=%" (ECHO publishing && CALL node_modules\.bin\node-pre-gyp --msvs_version=%msvs_version% publish %TOOLSET_ARGS% --runtime=electron --target=1.7.11 --disturl=https://atom.io/download/electron) ELSE (ECHO not publishing)
 IF %ERRORLEVEL% NEQ 0 GOTO ERROR
 
+
+CALL node_modules\.bin\node-pre-gyp rebuild package %TOOLSET_ARGS% --runtime=electron --target=1.8.6 --disturl=https://atom.io/download/electron
+SET CM=%APPVEYOR_REPO_COMMIT_MESSAGE%
+IF NOT "%CM%" == "%CM:[publish binary]=%" (ECHO publishing && CALL node_modules\.bin\node-pre-gyp --msvs_version=%msvs_version% publish %TOOLSET_ARGS% --runtime=electron --target=1.8.6 --disturl=https://atom.io/download/electron) ELSE (ECHO not publishing)
+IF %ERRORLEVEL% NEQ 0 GOTO ERROR
+
+CALL node_modules\.bin\node-pre-gyp rebuild package %TOOLSET_ARGS% --runtime=electron --target=2.0.4 --disturl=https://atom.io/download/electron
+SET CM=%APPVEYOR_REPO_COMMIT_MESSAGE%
+IF NOT "%CM%" == "%CM:[publish binary]=%" (ECHO publishing && CALL node_modules\.bin\node-pre-gyp --msvs_version=%msvs_version% publish %TOOLSET_ARGS% --runtime=electron --target=2.0.4 --disturl=https://atom.io/download/electron) ELSE (ECHO not publishing)
+IF %ERRORLEVEL% NEQ 0 GOTO ERROR
 
 GOTO DONE
 

